### PR TITLE
Clean up UI by using React Router DOM

### PR DESCRIPTION
**Pull request by dragosiordachioaia**
# Each section is now its own separate page.

I've basically split the code into separate components, and put each on a different <Route>

## Home:

diagrams-home

## Create Diagram:

diagrams-create-diagram

## Editor:

diagrams-editor

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -2,12 +2,16 @@ import React from "react";
...
 import DiagramEditor from "./DiagramEditor/DiagramEditor.jsx";
 
-const API_ID = "j6mw8j4i6h";
+import { Route, Switch, withRouter } from "react-router-dom";
 
-export default class App extends React.Component {
+import CreateDiagram from "./CreateDiagram/CreateDiagram";
+import DiagramList from "./DiagramList/DiagramList";
+
+const API_ID = "1vt7e367o8";
```
Comment (mec07) [inline frontend/src/App.js:10]: what is this? is it a secret?

File: frontend/src/App.js [hunk 2]
```diff
@@ -2,12 +2,16 @@ import React from "react";
...
+
+export class App extends React.Component {
   socket = undefined;
   state = {
-    diagramName: "",
```
Comment (mec07) [inline frontend/src/App.js:10]: you've removed "diagramName", but then set it in line 131

File: frontend/src/App.js [hunk 3]
```diff
@@ -118,20 +122,28 @@ export default class App extends React.Component {
...
     this.socket.send(
-      JSON.stringify({ message: "creatediagram", data: this.state.diagramName })
+      JSON.stringify({ message: "creatediagram", data: diagramName })
     );
     this.setState({
       diagramName: "",
-      diagrams: [...this.state.diagrams, this.state.diagramName],
+      diagrams: [...diagrams, diagramName],
     });
+    this.props.history.push("/");
```
Comment (mec07) [inline frontend/src/App.js]: what does this do?

Comment (mec07) [reply frontend/src/App.js]: don't we want to redirect to `/diagrams/$[diagramName}`?

File: frontend/src/App.js (hunk 1 above)
Comment (dragosiordachioaia) [reply frontend/src/App.js:10]: it's not a secret, it's part of the API URL

File: frontend/src/DiagramEditor/DiagramEditor.jsx [hunk 4]
```diff
@@ -129,28 +134,38 @@ export default class DiagramEditor extends React.Component {
     });
   };
 
-  onMouseWheel = (e) => {
+  zoom = (e) => {
```
Comment (mec07) [inline frontend/src/DiagramEditor/DiagramEditor.jsx:137]: I'm guessing that you're not done with this fuction as there are several commented lines and a console log...

File: frontend/src/DiagramList/DiagramList.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    return diagrams.map((diagramId) => (
+      <Link
+        to={`/diagrams/${diagramId}`}
+        key={diagramId}
+        className="diagram-item"
+      >
+        <h3 className="title">{diagramId}</h3>
+        <p className="created">Created on: 01 Feb 2020</p>
```
Comment (mec07) [inline frontend/src/DiagramList/DiagramList.jsx:23]: do diagrams not a date created attribute?

Comment (mec07) [reply frontend/src/DiagramList/DiagramList.jsx:23]: maybe we also want, last edited date?

**Review by mec07 — COMMENTED** [review]
Generally looks good, but I'd like to discuss it before approving it

File: frontend/src/App.js (hunk 3 above)
Comment (dragosiordachioaia) [reply frontend/src/App.js]: I was thinking to go to the list of diagrams, but it does make sense to go to the diagram's page. I'll change it now.

File: frontend/src/DiagramEditor/DiagramEditor.jsx (hunk 4 above)
Comment (dragosiordachioaia) [reply frontend/src/DiagramEditor/DiagramEditor.jsx:137]: it's not quite done, no. But I'd rather leave that code in for now, until I can get around to finishing it

File: frontend/src/DiagramList/DiagramList.jsx (hunk 5 above)
Comment (dragosiordachioaia) [reply frontend/src/DiagramList/DiagramList.jsx:23]: they don't have one for now, I'll add one as part of a separate task

**Review by mec07 — APPROVED** [review]
LGTM

**Pull request merged**